portal/common/lib/page_fetching.ts

```diff
@@ -18,166 +18,176 @@ import {
 import { aggregatorEndpoint } from "./aggregator";
 import { toBase64 } from "@mysten/bcs";
 import { sha256 } from "./crypto";
-import { getRoutes, matchPathToRoute } from "./routing";
+import { WalrusSitesRouter } from "./routing";
 import { HttpStatusCodes } from "./http/http_status_codes";
 import logger from "./logger";
 import BlocklistChecker from "./blocklist_checker";
 
 /**
- * Resolves the subdomain to an object ID, and gets the corresponding resources.
- *
- * The `resolvedObjectId` variable is the object ID of the site that was previously resolved. If
- * `null`, the object ID is resolved again.
- */
-export async function resolveAndFetchPage(
-    parsedUrl: DomainDetails,
-    resolvedObjectId: string | null,
-    blocklistChecker?: BlocklistChecker
-): Promise<Response> {
-    logger.debug({ message: "parsed-url", subdomain: parsedUrl.subdomain, path: parsedUrl.path });
-    if (!resolvedObjectId) {
-        const resolveObjectResult = await resolveObjectId(parsedUrl);
-        const isObjectId = typeof resolveObjectResult == "string";
-        if (!isObjectId) {
-            return resolveObjectResult;
+* Includes all the logic for fetching a walrus site.
```

Comment:
Suggestion: see the previous comment
```suggestion
* Includes all the logic for fetching a walrus URL.
```